src/per/pin.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
```

Comment:
I don't see anything that uses this, and it seems like this some how got tied in or checked out from #354 

If it's not in use let's remove it until we work on that PR.